### ISSUE TITLE
[Bug] Cannot take screenshot of videos running in PiP window from private tab

### ISSUE THREAD

**Issue by botistreamer** [mozilla-mobile/fenix#20098]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
1. Enable `Allow screenshots in private browsing` from settings
2. Open private tab
3. Open any video
4. Enter fullscreen
5. Press home button to enter PiP mode
6. Take a screenshot

### Expected behavior
Take screenshot as usual

_to see expected behavior follow same steps but in a normal tab_
[screen recording Imgur link](https://imgur.com/a/4fq0okj)
### Actual behavior
1. `Couldn't save screenshot`
2. **Screen recording of PiP windows appears blank** instead of the content like in normal tab. (Content on my actual device screen is **not blank** )
[screen recording Imgur link](https://imgur.com/a/s1rvzrs)
### Device information

* Device vendor / model and Android version: Poco X3, Android 11, ArrowOS
* Firefox for Android version: 89.1.1

**Comment by jonalmeida** [comment]
Possibly related to https://github.com/mozilla-mobile/fenix/commit/0572cab97c9e1e6b83c6e553f97531940b0cf38b which is the last place we modified for setting secure flags on fragments/activities?

**Comment by lobontiumira** [comment]
I can confirm the issue on all the builds, 6/22 Nightly 91.0a1, Beta 90.0.0-beta.4, and RC 89.1.1, with Google Pixel (Android 10).

**Comment by s3xycake** [comment]
I can reproduce this issue on Firefox Nightly 91.0a1 (Build #2015818075) on a One Plus 8 pro (Android 11).

**Comment by codrut-topliceanu** [comment]
> Possibly related to [0572cab](https://github.com/mozilla-mobile/fenix/commit/0572cab97c9e1e6b83c6e553f97531940b0cf38b) which is the last place we modified for setting secure flags on fragments/activities?

There was an issue here, indeed. 
But there's more to it than that, PiP has the flag locked somehow.

**Comment by codrut-topliceanu** [comment]
I asked for some UX-feedback for this issue on a related meta https://github.com/mozilla-mobile/fenix/issues/20202 . I'm trying to centralise the information in one issue.

**Comment by Nickoriginal** [comment]
>When `Allow screenshots in private browsing` is **Disabled** (settings->private browsing) (image1):
-- Fenix private browsing videos in Picture in Picture mode should not be screenshot-able? -> **Needs UX feedback**

>When `Allow screenshots in private browsing` is **Enabled** (settings->private browsing) (image1):
-- Fenix private browsing videos in Picture in Picture mode should be screenshot-able? -> **Needs UX feedback**

I think that, as you wrote, Fenix private browsing videos in PiP mode should be **screenshotable**, when `Allow screenshots in private browsing` is enabled and **not screenshotable**, when `Allow screenshots in private browsing` is disabled. In my opinion, this is the most correct decision.

**Comment by amedyne** [comment]
@topotropic Would be good to get your input on this one. Thanks!

**Comment by topotropic** [comment]
Yes, I agree – when people allow screenshots in private mode, they should also be able to take a  screenshot when they view a video in PIP. Thanks!

**Comment by codrut-topliceanu** [comment]
Talked with @topotropic on Slack. Waiting on input from @vesta0

**Comment by vesta0** [comment]
Discussed this with @topotropic a couple of weeks ago and we agreed that this is a bug that needs to be fixed. Allowing screenshots should also apply to PiP mode.

**Comment by codrut-topliceanu** [comment]
Aweaome! Thank you ❤️ ! I'll get started on a PR asap.

**Comment by codrut-topliceanu** [comment]
> Discussed this with @topotropic a couple of weeks ago and we agreed that this is a bug that needs to be fixed. Allowing screenshots should also apply to PiP mode.

@vesta0 As I've said to @topotropic on Slack, the fix for this issue will make it so that `Allow screenshots in private browsing` TRUE will no longer hide the contents of Private Browsing Fenix in Recents App. I hope that is agreeable. I will have a PR up soon.

**Comment by vesta0** [comment]
Yes that is fine @codrut-topliceanu thanks for confirming.

**Comment by SoftVision-LorandJanos** [comment]
Verified as fixed in the latest Nightly (93.0a1 20210819T05:09:47.445098).

Devices used:

- Xiaomi Mi11 Lite (Android 11)
- Samsung Galaxy Tab 3 (Android 9).
